cli/tests/test_git_push.rs

```diff
@@ -2085,17 +2093,20 @@ fn test_git_push_sign_on_push() {
     insta::assert_snapshot!(output, @r"
     ------- stderr -------
     Updated signatures of 2 commits
-    Rebased 2 descendant commits
+    Rebased 3 descendant commits
     Changes to push to origin:
-      Move forward bookmark bookmark2 from 38a204733702 to d45e2adce0ad
-    Working copy  (@) now at: kmkuslsw 3d5a9465 (empty) commit which should not be signed 2
-    Parent commit (@-)      : kpqxywon 48ea83e9 (empty) commit which should not be signed 1
+      Move forward bookmark bookmark2 from 38a204733702 to 87d9bf2ca4fb
+    Working copy  (@) now at: wqnwkozp eb60bb86 (empty) commit which is signed initially
+    Parent commit (@-)      : kmkuslsw 50a620ed (empty) commit which should not be signed 2
     [EOF]
     ");
-    // Only commits which are being pushed should be signed
+    // Commits which are being pushed should be signed.
+    // The initially signed commit should be resigned if its ancestors are signed.
```

Comment:
To test this, we'll need to turn the default signing off: `signing.behavior = "drop"`

That said, the next patch moves it to user's problem. The example configuration might have to be something like `roots(~signed()):: & mine()` (or `roots(mutable() & ~signed()):: & mine()` to not check signature of uninteresting commits?)

Reply:
> To test this, we'll need to turn the default signing off: `signing.behavior = "drop"`

I might be misunderstanding, but If the default signing behavior is `drop`, then the initially signed commit should not be resigned if its ancestors are signed when pushing.

Reply:
Thanks, I think I misunderstood your original comment. You're referring to the issue where given:

```
A (not signed) -> B (signed)
```

 and B is to be pushed, the original code before this PR could lead to B not being signed if `signing.behavior = "drop"`.

I was testing a different case where:

```
A (not signed) -> B (not signed) -> C (signed)
```

and B is to be pushed, C should still be resigned after A and B are signed if `signing.behavior = "keep"`.

I think I might rewrite the tests to ensure both cases are tested.